File: src/main/scala/com/fillta/higgs/boson/demo/PoloExample.scala
package com.fillta.higgs.boson.demo

import info.crlog.higgs.protocols.boson.v1.BosonProperty

case class PoloExample() {
  def this(j: Int) = {
    this()
    i = j
  }

  @BosonProperty
  var i = 0
  var name = "Test non-annotated field"
  private var str = "Test private non-annotated field"
  @BosonProperty(ignore = true)
  var ignored: String = null

  var nested = new Nested()

  override def toString = {
    val buf = new StringBuilder()
    buf.append("[")
    for (field <- getClass().getDeclaredFields()) {
      field.setAccessible(true)
      val value = field.get(this)
      if (field.getType().isArray) {
        buf.append(field.getName()).append(" = ")
        for (v <- value.asInstanceOf[Array[_]]) {
          buf.append(value).append(",")
        }
      } else {
        buf.append(field.getName()).append(" = ").append(value)
      }
      buf.append(",\n")
    }
    buf.append("]")
    buf.mkString
  }
}

class Nested {
  var array = Array(new NestedField(), new NestedField(), new NestedField())
  var list = List("a", "b", "c", "d")

  override def toString = {
    val buf = new StringBuilder()
    buf.append("[")
    for (field <- getClass().getDeclaredFields()) {
      field.setAccessible(true)
      val value = field.get(this)
      if (field.getType().isArray) {
        buf.append(field.getName()).append(" = ")
        for (v <- value.asInstanceOf[Array[_]]) {
          buf.append(value).append(",")
        }
      } else {
        buf.append(field.getName()).append(" = ").append(value)
      }
      buf.append(",\n")
    }
    buf.append("]")
    buf.mkString
  }
}

class NestedField {
  var a = 0
  var b = 0L
  var c = 0D
  var d = 0F
  var map = Map("0 dim" -> "map", "sum" -> "bing bong", "bah" -> 12345)

  override def toString = {
    val buf = new StringBuilder()
    buf.append("[")
    for (field <- getClass().getDeclaredFields()) {
      field.setAccessible(true)
      val value = field.get(this)
      if (field.getType().isArray) {
        buf.append(field.getName()).append(" = ")
        for (v <- value.asInstanceOf[Array[_]]) {
          buf.append(value).append(",")
        }
      } else {
        buf.append(field.getName()).append(" = ").append(value)
      }
      buf.append(",\n")
    }
    buf.append("]")
    buf.mkString
  }
}

File: examples/src/main/java/io/higgs/examples/boson/PoloExample.java
package io.higgs.examples.boson;

import io.higgs.boson.serialization.BosonProperty;

import java.lang.reflect.Field;

public class PoloExample {
    public static boolean dont = true;
    public static boolean stillDont;
    @BosonProperty
    int i;
    String name = "Test non-annotated field";
    @BosonProperty(ignore = true)
    String ignored;
    Nested nested = new Nested();
    private String str = "Test private non-annotated field";

    public PoloExample() {
    }

    public PoloExample(int j) {
        i = j;
    }

    public String toString() {
        StringBuilder buf = new StringBuilder();
        buf.append("[");
        for (Field field : getClass().getDeclaredFields()) {
            field.setAccessible(true);
            Object value = null;
            try {
                value = field.get(this);
            } catch (IllegalAccessException e) {
                //bleh!
            }
            if (field.getType().isArray()) {
                buf.append(field.getName()).append(" = ");
                for (Object v : (Object[]) value) {
                    buf.append(v).append(",");
                }
            } else {
                buf.append(field.getName()).append(" = ").append(value);
            }
            buf.append(",\n");
        }
        buf.append("]");
        return buf.toString();
    }

}
